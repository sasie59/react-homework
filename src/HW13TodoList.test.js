import React from 'react';
import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import HW from './HW13TodoList';

test('should HW13TodoList', () => {
  const { container } = render(<HW />);
  expect(container).toMatchSnapshot();
});

test('should add something to list', () => {
  const { container } = render(<HW />);
  const input = container.querySelector('input');
  userEvent.type(input, "aaa{enter}");
  expect(container).toMatchSnapshot();
});

