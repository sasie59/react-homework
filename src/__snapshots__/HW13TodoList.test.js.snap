// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should HW13TodoList 1`] = `
<div>
  <div>
    <h1>
      HW13TodoList
    </h1>
    <h1>
      代辦事項
    </h1>
    <form>
      <input
        placeholder="add something ?"
        type="text"
        value=""
      />
      <button>
        save
      </button>
    </form>
  </div>
</div>
`;

exports[`should add something to list 1`] = `
<div>
  <div>
    <h1>
      HW13TodoList
    </h1>
    <h1>
      代辦事項
    </h1>
    <form>
      <input
        placeholder="add something ?"
        type="text"
        value=""
      />
      <button>
        save
      </button>
    </form>
    <div>
      <label
        class=""
      />
    </div>
    <div>
      <label
        class=""
      >
        aaa
      </label>
    </div>
  </div>
</div>
`;
